books/tests.py

```diff
@@ -137,3 +140,77 @@ def test_can_create_book_with_cc_license(self):
                     )
         book_index.add_child(instance=book)
         self.assertEqual(book.license_url, 'https://creativecommons.org/licenses/by/4.0/')
+
+
+class ResourceTests(WagtailPageTests):
+    def setUp(self):
+        pass
+
+    @classmethod
+    def setUpTestData(cls):
+        # create root page
+        root_page = Page.objects.get(title="Root")
+        # create homepage
+        homepage = HomePage(title="Hello World",
+                            slug="hello-world",
+                            )
+        # add homepage to root page
+        root_page.add_child(instance=homepage)
+        # create book index page
+        book_index = BookIndex(title="Book Index",
+                               page_description="Test",
+                               dev_standard_1_description="Test",
+                               dev_standard_2_description="Test",
+                               dev_standard_3_description="Test",
+                               dev_standard_4_description="Test",
+                               )
+        # add book index to homepage
+        homepage.add_child(instance=book_index)
+
+        test_image = SimpleUploadedFile(name='openstax.png',
+                                        content=open("oxauth/static/images/openstax.png", 'rb').read())
+        cls.test_doc = Document.objects.create(title='Test Doc', file=test_image)
+
+        cls.book_index = Page.objects.get(id=book_index.id)
+
+    def test_faculty_resources_available(self):
+        book_index = BookIndex.objects.all()[0]
```

Comment:
you should be able to do `cls.book_index` although I usually do all the things you have in `setupTestData` in `setUp` and use `self`. I'm not sure I've used `setupTestData`.
Either way, you shouldn't need to fetch the created pages from the database since you created it in during setup.